src/contracts/Qearn.h

```diff
@@ -0,0 +1,859 @@
+using namespace QPI;
+
+constexpr uint64 QEARN_MINIMUM_LOCKING_AMOUNT = 10000000;
+constexpr uint64 QEARN_MAX_LOCKS = 4194304;
+constexpr uint64 QEARN_MAX_EPOCHS = 4096;
+constexpr uint64 QEARN_MAX_USERS = 131072;
+constexpr uint64 QEARN_MAX_LOCK_AMOUNT = 1000000000000ULL;
+constexpr uint64 QEARN_INITIAL_EPOCH = 128;                             //  we need to change this epoch when merging
+
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_0_3 = 0;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_4_7 = 5;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_8_11 = 5;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_12_15 = 10;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_16_19 = 15;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_20_23 = 20;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_24_27 = 25;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_28_31 = 30;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_32_35 = 35;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_36_39 = 40;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_40_43 = 45;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_44_47 = 50;
+constexpr uint64 QEARN_EARLY_UNLOCKING_PERCENT_48_51 = 55;
+
+constexpr uint64 QEARN_BURN_PERCENT_0_3 = 0;
+constexpr uint64 QEARN_BURN_PERCENT_4_7 = 45;
+constexpr uint64 QEARN_BURN_PERCENT_8_11 = 45;
+constexpr uint64 QEARN_BURN_PERCENT_12_15 = 45;
+constexpr uint64 QEARN_BURN_PERCENT_16_19 = 40;
+constexpr uint64 QEARN_BURN_PERCENT_20_23 = 40;
+constexpr uint64 QEARN_BURN_PERCENT_24_27 = 35;
+constexpr uint64 QEARN_BURN_PERCENT_28_31 = 35;
+constexpr uint64 QEARN_BURN_PERCENT_32_35 = 35;
+constexpr uint64 QEARN_BURN_PERCENT_36_39 = 30;
+constexpr uint64 QEARN_BURN_PERCENT_40_43 = 30;
+constexpr uint64 QEARN_BURN_PERCENT_44_47 = 30;
+constexpr uint64 QEARN_BURN_PERCENT_48_51 = 25;
+
+constexpr sint32 QEARN_INVALID_INPUT_AMOUNT = -1;
+constexpr sint32 QEARN_LOCK_SUCCESS = 0;
```

Comment:
I recommend using non-zero for success, because default is zero. If lock is called from another contract with a negative amount, the output will be 0, indicating success although the lock procedure is not called at all. It would make sense to use QEARN_INVALID_INPUT_AMOUNT = 0.